### PR TITLE
docs: Update state store code examples

### PR DESCRIPTION
**Pull request by szuperaz**
- Remove TODOs from state store docs
- Update code examples to use `StreamMeeting` instead of `StreamCallProvider`

### CODE REVIEW

File: packages/react-sdk/docusaurus/docs/React/02-guides/03-call-and-participant-state.mdx [hunk 1]
```diff
@@ -3,89 +3,193 @@ title: Call & Participant State
...
+export default function App() {
+  const client = useCreateStreamVideoClient({
+    apiKey: '<Your API key>',
+    tokenOrProvider: '<token provider>',
+    user: {
+      id: '<user id>',
+      name: '<name>',
+    },
+  });
+  return ();
```
Comment (oliverlaz) [inline packages/react-sdk/docusaurus/docs/React/02-guides/03-call-and-participant-state.mdx:32]:
```suggestion
  return <StreamVideo client={client}><MyApp /></StreamVideoClient>;
```

File: packages/react-sdk/docusaurus/docs/React/02-guides/03-call-and-participant-state.mdx [hunk 2]
```diff
@@ -3,89 +3,193 @@ title: Call & Participant State
...
+        data={{ create: true }}
+      ></StreamMeeting>
+    </StreamVideo>
+  );
+}
+
+const Header = () => {
+  const user = useConnectedUser();
+  return <div>{user ? `Logged in: ${user.name}` : 'Logged out'}</div>;
+};
```
Comment (oliverlaz) [inline packages/react-sdk/docusaurus/docs/React/02-guides/03-call-and-participant-state.mdx:136]:
```suggestion
  return (
    <StreamVideo client={client}>
      <StreamMeeting
        callType="default"
        callId="test-call"
        autoJoin={true}
        data={{ create: true }}
      >
        <Header />
      </StreamMeeting>
    </StreamVideo>
  );
}

const Header = () => {
  const user = useConnectedUser();
  const recording = useIsCallRecordingInProgress()
  return (
    <div>
      { recording && <RecordingBadge />}
      {user ? `Logged in: ${user.name}` : 'Logged out'}
    </div>
  );
};
```

Comment (oliverlaz) [inline packages/react-sdk/docusaurus/docs/React/02-guides/03-call-and-participant-state.mdx:136]:
```suggestion
  return (
    <StreamVideo client={client}>
      <StreamMeeting
        callType="default"
        callId="test-call"
        autoJoin={true}
        data={{ create: true }}
      >
        <Header />
      </StreamMeeting>
    </StreamVideo>
  );
}

const Header = () => {
  const user = useConnectedUser();
  const recording = useIsCallRecordingInProgress()
  return (
    <div>
      { recording && <RecordingBadge />}
      {user ? `Logged in: ${user.name}` : 'Logged out'}
    </div>
  );
};
```

File: packages/react-sdk/docusaurus/docs/React/02-guides/03-call-and-participant-state.mdx (hunk 1 above)
Comment (szuperaz) [reply packages/react-sdk/docusaurus/docs/React/02-guides/03-call-and-participant-state.mdx:32]: The `StreamVideo` context is added in the next step

File: packages/react-sdk/docusaurus/docs/React/02-guides/03-call-and-participant-state.mdx (hunk 2 above)
Comment (szuperaz) [reply packages/react-sdk/docusaurus/docs/React/02-guides/03-call-and-participant-state.mdx:136]: The next step will introduce the `useCall` and `useCallingState` hooks, I can also include the `useIsCallRecordingInProgress` hook if you want to, but I'd rather add it to the `Call` component

**Pull request merged**